lib/config.js

```diff
@@ -44,6 +44,49 @@ function putNestedValue(params, key, value) {
   return params;
 }
 
+function parseCloudinaryConfigFromEnvURL(ENV_STR) {
+  let conf = {};
+
+  let uri = url.parse(ENV_STR, true);
+
+  if (uri.protocol === 'cloudinary:') {
+    conf = Object.assign({}, conf, {
+      cloud_name: uri.host,
```

Comment:
I would extract common parsing to a separate function and reuse it in `account:` and `cloudinary:` as follows:

generic function parseURL(url)

returns:

{
    host: ...
    user: ...
    pass: ...
    path: ...(without leading char)
}

And  after that you just map `host` to `cloud_name` or `account_id` accordingly, `user` and `pass` as well, and it can be further reused if we have more types of config URLs.

Reply:
I tried to avoid over-abstraction here, do you really think this refactor is also needed as part of this feature?